sw/device/lib/testing/otp_ctrl_testutils.c

```diff
@@ -0,0 +1,43 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/*
+ * OTP the Direct Access Interface (DAI) operation time-out in micro seconds.
+ *
+ * It is not possible to predict the specific cycle count that a DAI operation
+ * takes, thus arbitrary value of 100us is used.
+ */
+const uint8_t kOtpDaiTimeoutUs = 100;
+
+/**
+ * Checks whether the DAI operation has finished.
+ */
+static bool dai_finished(const otp_ctrl_t *otp) {
+  dif_otp_ctrl_status_t status;
+  CHECK_DIF_OK(dif_otp_ctrl_get_status(&otp->dif, &status));
+  return bitfield_bit32_read(status.codes, kDifOtpCtrlStatusCodeDaiIdle);
+}
+
+otp_ctrl_t otp_ctrl_init(void) {
+  otp_ctrl_t otp;
+
+  mmio_region_t region =
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR);
+  CHECK_DIF_OK(dif_otp_ctrl_init(region, &otp.dif));
+
+  return otp;
+}
+
+void otp_wait_for_dai(const otp_ctrl_t *otp) {
+  IBEX_SPIN_FOR(dai_finished(otp), kOtpDaiTimeoutUs);
+}
```

Comment:
Same here. Why not just call `IBEX_SPIN_FOR` in the test that needs it? Why do we need an extra layer of abstraction for this?

Reply:
Well, one reason being that N users don't need to care what the time-out value should be set as - if it ever trips for some reason, there is a single point where we would have to change it.

Reply:
If you guys think that it is advantageous to be able to set up timeout separately in individual tests, then I can change this to use `IBEX_SPIN_FOR` inside the individual tests. Otherwise it is just simpler if you don't need to know anything apart form calling this testutil.